crates/runtime/src/dialogue.rs

```diff
@@ -253,12 +183,10 @@ impl Dialogue {
     /// The original states that the [`LineHandler`] and [`CommandHandler`] may call [`Dialogue::continue_`]. Because of the borrow checker,
     /// this is action is very unidiomatic and impossible to do without introducing a lot of interior mutability all along the API.
     /// For this reason, we disallow mutating the [`Dialogue`] within any handler.
-    pub fn continue_(&mut self) -> &mut Self {
+    #[must_use]
+    pub fn continue_(&mut self) -> Option<Vec<DialogueEvent>> {
```

Comment:
Would this work?
```suggestion
    pub fn r#continue(&mut self) -> Option<Vec<DialogueEvent>> {
```

Reply:
It would, but in my experience, it doesn't play nice with IDEs. Counter suggestion: implement `Iterator` and call it `next()` :)

Reply:
[`e885283` (#103)](https://github.com/yarn-slinger/yarn-slinger/pull/103/commits/e885283a9c369a29a3cf3c315a5689c214a54059)